test/drivers/test_cqrrp_gpu.cu

```diff
@@ -0,0 +1,358 @@
+
+        T atol1 = std::pow(std::numeric_limits<T>::epsilon(), 0.75);
+        T atol2 = std::pow(std::numeric_limits<T>::epsilon(), 0.60);
+        ASSERT_NEAR(col_nrm_J,   0.0, atol1);
+        ASSERT_NEAR(col_nrm_tau, 0.0, atol1);
+        ASSERT_NEAR(norm_R_diff, 0.0, atol2);
+    
+    	cudaError_t ierr = cudaGetLastError();
+    	if (ierr != cudaSuccess)
+    	{
+        	RandLAPACK_CUDA_ERROR("Error before test_CQRRP_compare_with_CPU returned. " << cudaGetErrorString(ierr))
+        	abort();
+    	}
+    }
+};
+
+// Note: If Subprocess killed exception -> reload vscode
+TEST_F(TestCQRRP, CQRRP_GPU_070824) {
+    int64_t m = 5000;//5000;
+    int64_t n = 2800;//2000;
+    int64_t k = 2800;
+    double d_factor = 1;//1.0;
+    int64_t b_sz = 900;//500;
+    int64_t d = d_factor * b_sz;
+    double norm_A = 0;
+    double tol = std::pow(std::numeric_limits<double>::epsilon(), 0.85);
+    auto state = RandBLAS::RNGState();
+    bool profile_runtime = true;
+
+    CQRRPTestData<double> all_data(m, n, k, d);
+    RandLAPACK::CQRRP_blocked_GPU<double, r123::Philox4x32> CQRRP_blocked_GPU(profile_runtime, tol, b_sz);
+    CQRRP_blocked_GPU.use_qrf = false;
+
+    RandLAPACK::gen::mat_gen_info<double> m_info(m, n, RandLAPACK::gen::gaussian);
+    //RandLAPACK::gen::mat_gen_info<double> m_info(m, n, RandLAPACK::gen::polynomial);
+    //m_info.cond_num = 2;
+    //m_info.rank = k;
+    //m_info.exponent = 2.0;
+    RandLAPACK::gen::mat_gen<double, r123::Philox4x32>(m_info, all_data.A.data(), state);
+
+    norm__sektch_and_copy_computational_helper<double, r123::Philox4x32>(norm_A, d, all_data, state);
+#if !defined(__APPLE__)
+    test_CQRRP_general<double, RandLAPACK::CQRRP_blocked_GPU<double, r123::Philox4x32>>(d, norm_A, all_data, CQRRP_blocked_GPU);
+#endif
+}
+
+// Note: If Subprocess killed exception -> reload vscode
+TEST_F(TestCQRRP, CQRRP_GPU_vectors) {
+    int64_t m = 5000;//5000;
+    int64_t n = 2800;//2000;
+    int64_t k = 2800;
+    double d_factor = 1;//1.0;
+    int64_t b_sz = 900;//500;
+    int64_t d = d_factor * b_sz;
+    double norm_A = 0;
+    double tol = std::pow(std::numeric_limits<double>::epsilon(), 0.85);
+    auto state = RandBLAS::RNGState();
+
+    CQRRPTestData<double> all_data(m, n, k, d);
+    RandLAPACK::CQRRP_blocked_GPU<double, r123::Philox4x32> CQRRP_blocked_GPU(false, tol, b_sz);
+    RandLAPACK::CQRRP_blocked<double, r123::Philox4x32> CQRRP_blocked_CPU(false, tol, b_sz);
+    CQRRP_blocked_CPU.nnz = 2;
+    CQRRP_blocked_CPU.num_threads = 4;
+    CQRRP_blocked_CPU.use_gaussian = true;
+
+    RandLAPACK::gen::mat_gen_info<double> m_info(m, n, RandLAPACK::gen::gaussian);
+    RandLAPACK::gen::mat_gen<double, r123::Philox4x32>(m_info, all_data.A.data(), state);
+
+    norm__sektch_and_copy_computational_helper<double, r123::Philox4x32>(norm_A, d, all_data, state);
+#if !defined(__APPLE__)
+    test_CQRRP_compare_with_CPU(d, norm_A, all_data, CQRRP_blocked_GPU, CQRRP_blocked_CPU, state);
+#endif
```

Comment:
Have this test be in a more prominent place. I think the whole file should be skipped if we're on macOS right now, right?

Reply:
Yeah. Fixed.